src/python/tests/core/bot/minimizer/js_minimizer_test.py

```diff
@@ -0,0 +1,167 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for the Antlr Tokenizer."""
+
+from bot.minimizer import js_minimizer
+from bot.tokenizer import antlr_tokenizer
+from bot.tokenizer.grammars.JavaScriptLexer import JavaScriptLexer
+from tests.test_libs import helpers
+import unittest
+
+
+class JSMinimizerTest(unittest.TestCase):
+  """Test for Javascript Minimizer"""
+
+  def setUp(self):
+    self._tests_to_queue = []
+
+    helpers.patch_environ(self)
+    helpers.patch(self, [
+        'bot.minimizer.minimizer.Testcase.prepare_test',
+        'bot.minimizer.minimizer.Testcase.process'
+    ])
+
+    helpers.patch_environ(self)
+    self.mock.prepare_test.side_effect = self._mock_prepare_test
+    self.mock.process.side_effect = self._mock_process()
+
+    self._tokenizer = antlr_tokenizer.AntlrTokenizer(JavaScriptLexer)
+    self._minimizer = js_minimizer.JSMinimizer(
+        self._mock_test_function(),
+        tokenizer=self._tokenizer.tokenize,
+        token_combiner=self._tokenizer.combine)
+
+  def _mock_process(self):
+    return
+
+  def _mock_prepare_test(self, testcase, hypothesis):
+    txt = ""
+    for index in hypothesis:
+      txt += testcase.tokens[index]
+
+    self._tests_to_queue.append(txt)
+
+  def _mock_test_function(self):
+    return True
+
+  def test_minimize_empty_string(self):
+    data = ""
+
+    self._minimizer.minimize(data)
+
+    self.assertEqual(self._tests_to_queue, [])
+
+  def test_bracket_hypotheses_1(self):
+    """Hypothesis is: Remove from start of line to open brace and the closing
+    brace. e.g.: if (statement_that_evaluates_to_true) { crash() } -> crash()"""
+    data = "if(boolean) { crash }"
+
+    self._minimizer.minimize(data)
+
+    self.assertIn("if(boolean) {}", self._tests_to_queue)
+
+  def test_bracket_hypotheses_2(self):
```

Comment:
Ideally these names should be meaningful without an understanding of the comments in the minimizer (e.g. this could be renamed to test_try_catch_hypothesis). Some of them might be a bit more complicated, but I'd prefer any attempt at a description to the current scheme.

Reply:
Done